api/v1alpha8/types.go

```diff
@@ -240,6 +240,12 @@ type ServerGroupFilter struct {
 	Name string `json:"name,omitempty"`
 }
 
+type ServerGroupRef struct {
+	// Name of the OpenStackServerGroup resource to be used.
+	// Must be in the same namespace as the resource(s) being provisioned.
+	Name string `json:"name"`
+}
```

Comment:
I think LocalObjectReference [1] should be used as a type [2].

Reply:
Would it be an issue that it specifies `omitempty`?

Reply:
Ugh, it probably would. Okay, the design of this field is good, we can change internals if we want to later.

Reply:
Discussion resolved